src/mlpack/core/util/program_doc.hpp

```diff
@@ -18,49 +20,108 @@ namespace util {
 /**
  * A static object whose constructor registers program documentation with the
  * IO class.  This should not be used outside of IO itself, and you should use
- * the PROGRAM_INFO() macro to declare these objects.  Only one ProgramDoc
- * object should ever exist.
+ * these BINDING_PNAME(), BINDING_SHORT_DESC(), BINDING_LONG_DESC(), 
+ * BINDING_EXAMPLE() and BINDING_SEE_ALSO() macros to declare these objects.
+ * Only correspond object should ever exist.
  *
  * @see core/util/io.hpp, mlpack::IO
  */
-class ProgramDoc
+class ProgramName
 {
  public:
   /**
-   * Construct a ProgramDoc object.  When constructed, it will register itself
-   * with IO, and when the user calls --help (or whatever the option is named
-   * for the given binding type), the given function that returns a std::string
-   * will be returned.
+   * Construct a ProgramName object.  When constructed, it will register itself
+   * with IO.  A fatal error will be thrown if more than one is constructed.
    *
-   * @param programName Short string representing the name of the program.
-   * @param shortDocumentation A short two-sentence description of the program,
+   * @param programName Name of the default module.
+   */
+  ProgramName(const std::string& programName);
+
+  /**
+   * Construct an empty ProgramName object.  (This is not meant to be used!)
+   */
+  ProgramName();
+  std::string programName;
```

Comment:
```suggestion
  ProgramName();
  
  std::string programName;
```
Just adding some spacing for style. :)